code/homophones.py

```diff
@@ -22,17 +22,35 @@
 mod.mode("homophones")
 mod.list("homophones_canonicals", desc="list of words ")
 
-
 main_screen = ui.main_screen()
 
 
-def update_homophones(name, flags):
-    if name != homophones_file:
-        return
+def get_homophones_from_csv(filename: str):
```

Comment:
Could this use get_list_from_csv as well? Superficially they look quite similar.

Reply:
Homophone's logic is slightly different for parsing the CSV. It builds 
- a single list with only the first element of each homophone listed. No remapping.
- separately builds a dictionary of all the homophones -> alternates for selection-based shenanigans

The rest of the logic is, of course, identical. I was a bit lazy since I didn't see a quick way to consolidate these, and didn't immediately see any use cases for this outside of homophones. Thoughts?

Reply:
Ah right, homophones isn't a CSV with a fixed number of columns I see. Probably not worth merging them in that case.